### DISCUSSION TITLE
Xbox controller setup?

### DISCUSSION THREAD

**Discussion by trueger** [pybricks/support#1612] [General]
I don't have any experience with game consoles, many apologies if this is a trivial ask.

Can I buy a spare Xbox controller and pair it out of the box with a Pybricks-programmed hub, or is there some additional controller setup needed where I'd need to use a controller from an Xbox system?

**Comment by laurensvalk** [comment]
You may need to update the Xbox Controller, but you can do so without the Xbox Console.

You can use the Xbox Accessories app on a Windows computer. I don't have an Xbox either  :smile: 

We've now included the instructions with each of our examples, e.g: https://pybricks.com/project/technic-42109-xbox/#updating-the-xbox-controller

**Comment by trueger** [comment]
Got it, many thanks!

I don't have a Windows PC either, but I do have an M1 Mac.

I found this note about how to update the controller via a VM, I may give it a try:  https://old.reddit.com/r/macgaming/comments/u4eunt/solution_how_to_update_xbox_controller_firmware/

**Reply by laurensvalk** [reply]
Since it's a quick one off procedure, maybe you can do the update on a work or friend's windows computer.

It's also possible that the controller you'll buy already has a recent enough version, but there isn't a way to know for sure I think.

**Reply by trueger** [reply]
Understood, many thanks for the advice!
